### DISCUSSION TITLE
[IntelliSense] Any plan for supporting Leptos - A Rust-Based Framework?

### DISCUSSION THREAD

**Discussion by aswinmurali-io** [tailwindlabs/tailwindcss#13157] [Ideas]
Leptos is a rust-based web framework for building web apps. Leptos supports typing html code via the `view!` macro. Currently, the Tailwind CSS vscode extension does not support type completion here. Do note that the `view!` macro lets us type html code directly inside a rust file. Kinda like a `.jsx` file. Any plans to support Leptos? Currently in the rust world, Leptos is a popular framework for web development but not having tailwind editor support sucks :(

**Comment by jtmueller** [comment]
This actually works as-is with the right configuration. I'm using the following `.vscode/settings.json`:

```json
{
    "emmet.includeLanguages": {
        "rust": "html",
        "*.rs": "html"
    },
    "tailwindCSS.includeLanguages": {
        "rust": "html",
        "*.rs": "html"
    },
    "files.associations": {
        "*.rs": "rust"
    },
    "editor.quickSuggestions": {
        "other": "on",
        "comments": "on",
        "strings": true
    },
    "css.validate": false
}
```

**Comment by Salman-Sali** [comment]
For anyone else, for whom its still not working, like me, please check if you have tailwind.config.js file in your main dir.

**Comment by DanarQ** [comment]
I made an HTML/RSX IntelliSense extension for Leptos.

You can check it out on GitHub:
https://github.com/DanarQ/leptos-html-rsx-intellisense

I haven’t published it to the VS Code Extension Marketplace yet, but I’ll try to do that soon.

**Reply by jmevel** [reply]
Hi, great initiative from you, thanks for your effort.

Have you ever heard of the [Language Server Protocol (LSP)](https://microsoft.github.io/language-server-protocol/) ?
If you could make your project an LSP instead of a VS Code extension, it would work on any [editor supporting LSP](https://microsoft.github.io/language-server-protocol/implementors/tools/).

I personally left VS Code a while ago, now using Zed and very happy with it. I'd love to be able to use your project in my IDE.
LSP is honestly becoming the new standard, I think you should have a look at it.

**Reply by DanarQ** [reply]
Hi, sorry for the delayed response. That's a great suggestion! I'd love to transition this project to an LSP so it can support other editors too. I will update you here once it's ready. Thank you!
